Document/0x05d-Testing-Data-Storage.md

```diff
@@ -697,4 +697,120 @@ For this reason all notification usage should be inspected for confidential or h
 
 #### Keyboard Cache
 
-When users enter information in input fields, the software automatically suggests data. This feature can be very useful for messaging apps. However, the keyboard cache may disclose sensitive information when the user selects an input field that takes this type of information.
+When users enter information into input fields, the keyboard software often provides suggestions based on previously entered data. This auto-completion feature can be very useful for messaging apps and other scenarios. However, by default, the Android keyboard may retain (or "cache") input history to offer suggestions and auto-completion. In contexts where sensitive data is entered (such as passwords or PINs), this caching behavior can inadvertently expose sensitive information.
+
+Apps can control this behavior by appropriately configuring the `inputType` attribute on text input fields. There are several ways to do this:
+
+**XML Layouts:**
+
+In the app's XML layout files (typically located in the `/res/layout` directory after unpacking the APK), you can define the input type directly in the `<EditText>` element using the `android:inputType` attribute. For example, setting the input type to `"textPassword"` automatically disables auto-suggestions and caching:
+
+```xml
+<EditText
+    android:id="@+id/password"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:hint="@string/password_hint"
+    android:inputType="textPassword" />
+```
+
+**Using the Traditional Android View System:**
+
+When creating input fields in code using the traditional Android view system, you can set the input type programmatically. For example, using an `EditText` in Kotlin:
+
+```kotlin
+val input = EditText(context).apply {
+    hint = "Enter PIN"
+    inputType = InputType.TYPE_CLASS_NUMBER or InputType.TYPE_NUMBER_VARIATION_PASSWORD
+}
+```
+
+**Using Jetpack Compose:**
```

Comment:
Should we also mention Jetpack Compose in MASTG-TEST-0258?

Reply:
the relevant APIs (`KeyboardOptions`) are there but I'll make an explicit mention